### DISCUSSION TITLE
How predict_many works?

### DISCUSSION THREAD

**Discussion by raul-parada** [online-ml/river#1298] [Q&A]
Hi all,

I'm interested to foresee more than one sample in horizon, for instance, 5 steps in the future. I've seen BayesianLinearRegression https://riverml.xyz/dev/api/linear-model/BayesianLinearRegression/ provides the option, however, I don't know how to use it. Should I insert a dictionary of the same numbers of samples as the number of steps to foresee?
Thanks,

**Comment by MaxHalford** [comment]
I think what you're looking for is the `time_series` module, and not `predict_many`. `predict_many` is there to process mini-batches of data, but not to do forecasting several timesteps ahead.

**Reply by raul-parada** [reply]
Does it perform well for non-periodic data? I would like to use it to foresee vehicular Cartesian coordinates. I only use latitude and longitude attributes.

**Reply by MaxHalford** [reply]
If there is no trend/structure in your data, then no it won't work well. But I don't know enough about your problem to know for sure. I suggest trying for yourself!
